monitor.go

```diff
@@ -0,0 +1,223 @@
+package main
+
+import (
+	"encoding/hex"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	raven "github.com/getsentry/raven-go"
+	"github.com/pborman/uuid"
+	"github.com/taskcluster/statsum"
+	tcauth "github.com/taskcluster/taskcluster-client-go/auth"
+)
+
+// NewMonitor creates a new monitor
```

Comment:
I think this comment serves no purpose. Could you elaborate on what a monitor is, whether it can be called concurrently, how it should be used, etc?